src/views/ModelPlan/TaskList/Basics/Milestones/index.tsx

```diff
@@ -0,0 +1,629 @@
+          performancePeriodEnds: formikValues.performancePeriodEnds,
+          wrapUpEnds: formikValues.wrapUpEnds,
+          highLevelNote: formikValues.highLevelNote,
+          phasedIn: formikValues.phasedIn,
+          phasedInNote: formikValues.phasedInNote
+        }
+      }
+    })
+      .then(response => {
+        if (!response?.errors) {
+          if (redirect === 'task-list') {
+            history.push(`/models/${modelId}/task-list`);
+          }
+        }
+      })
+      .catch(errors => {
+        formikRef?.current?.setErrors(errors);
+      });
+  };
+
+  return (
+    <MainContent className="margin-bottom-5">
+      <div className="grid-container">
+        <div className="tablet:grid-col-12">
+          <BreadcrumbBar variant="wrap">
+            <Breadcrumb>
+              <BreadcrumbLink asCustom={Link} to="/">
+                <span>{h('home')}</span>
+              </BreadcrumbLink>
+            </Breadcrumb>
+            <Breadcrumb>
+              <BreadcrumbLink
+                asCustom={Link}
+                to={`/models/${modelId}/task-list/`}
+              >
+                <span>{h('tasklistBreadcrumb')}</span>
+              </BreadcrumbLink>
+            </Breadcrumb>
+            <Breadcrumb current>{t('breadcrumb')}</Breadcrumb>
+          </BreadcrumbBar>
+          <PageHeading className="margin-top-4 margin-bottom-1">
+            {t('heading')}
+          </PageHeading>
+          <p
+            className="margin-top-0 margin-bottom-1 font-body-lg"
+            data-testid="model-plan-name"
+          >
+            <Trans i18nKey="modelPlanTaskList:subheading">
+              indexZero {modelName} indexTwo
+            </Trans>
+          </p>
+          <p className="margin-bottom-2 font-body-md line-height-sans-4">
+            {t('helpText')}
+          </p>
+
+          <Formik
+            initialValues={initialValues}
+            onSubmit={values => {
+              handleFormSubmit(values, 'task-list');
+            }}
+            enableReinitialize
+            validationSchema={planBasicsSchema.pageThreeSchema}
+            validateOnBlur={false}
+            validateOnChange={false}
+            validateOnMount={false}
+            innerRef={formikRef}
+          >
+            {(formikProps: FormikProps<PlanBasicsMilestoneTypes>) => {
+              const {
+                dirty,
+                errors,
+                handleSubmit,
+                isValid,
+                setErrors,
+                setFieldError,
+                setFieldValue,
+                validateField,
+                values
+              } = formikProps;
+              const flatErrors = flattenErrors(errors);
+              const handleOnchange = (e: any, field: string) => {
+                if (e === '') {
+                  return;
+                }
+                try {
+                  setErrors({});
+                  new Date(e).toISOString();
```

Comment:
Not sure why this line is here

Reply:
The try/catch statement is an error handler. It tries `new Date(e).toISOString();` to see if it can actually convert `e` to a Date. If it is fine, it continues through the function (setting the field value and validation the field). If it errors, it will go to the `catch`  section.

Reply:
Wouldn't the catch case trigger without that when trying to set the date in the state setter method?